nutype_macros/src/common/gen/traits.rs

```diff
@@ -228,3 +228,56 @@ pub fn gen_impl_trait_serde_deserialize(
         }
     }
 }
+
+pub fn gen_impl_trait_default(
+    type_name: &TypeName,
+    default_value: impl ToTokens,
+    has_validation: bool,
+) -> TokenStream {
+    let default_implementation = if has_validation {
+        let tp = type_name.to_string();
+        quote!(
+            Self::new(#default_value)
+                .unwrap_or_else(|err| {
+                    let tp = #tp;
+                    panic!("\nDefault value for type {tp} is invalid.\nERROR: {err}\n")
+                })
+        )
+    } else {
+        quote!(
+            Self::new(#default_value)
+        )
+    };
+
+    // Unfortunately it's not possible to guarantee at the compile time that the default value will always
+    // satisfy the validation rules.
+    // For this purpose we generate a unit test to verify this at run time.
+    let unit_test = if has_validation {
+        #[cfg(not(feature = "nutype_test"))]
```

Comment:
This is becoming a comment pattern, so consider coming up with a macro rule. E.g.

```rs
if_feature("nutype_test",
    then: quote!("Feature is ON"),
    else: quote!("Feature is OFF"),
)
```